api/v1alpha08/conversion.go

```diff
@@ -0,0 +1,132 @@
+	"sigs.k8s.io/yaml"
+
+	"github.com/kiegroup/kogito-serverless-operator/api/metadata"
+)
+
+var namingRegexp = regexp.MustCompile("^[a-z0-9](-?[a-z0-9])*$")
+var allowedCharsRegexp = regexp.MustCompile("[^-a-z0-9]")
+var startingDash = regexp.MustCompile("^-+")
+
+const (
+	// see https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
+	dash      = "-"
+	charLimit = 253
+)
+
+// FromCNCFWorkflow converts the given CNCF Serverless Workflow instance in a new KogitoServerlessWorkflow Custom Resource.
+func FromCNCFWorkflow(cncfWorkflow *cncfmodel.Workflow) (*KogitoServerlessWorkflow, error) {
+	if cncfWorkflow == nil {
+		return nil, errors.New("CNCF Workflow is nil")
+	}
+	workflowCR := &KogitoServerlessWorkflow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: extractName(cncfWorkflow),
+			Annotations: map[string]string{
+				metadata.ExpressionLang: string(cncfWorkflow.ExpressionLang),
+				metadata.Version:        cncfWorkflow.Version,
+				metadata.Description:    cncfWorkflow.Description,
+			},
+		},
+	}
+	workflowBytes, err := yaml.Marshal(cncfWorkflow)
+	if err != nil {
+		return nil, err
+	}
+	workflowCRFlow := &Flow{}
+	if err = yaml.Unmarshal(workflowBytes, workflowCRFlow); err != nil {
+		return nil, err
+	}
+	workflowCR.Spec.Flow = *workflowCRFlow
+
+	return workflowCR, nil
+}
+
+// ToCNCFWorkflow converts a KogitoServerlessWorkflow object to a Workflow one in order to be able to convert it to a YAML/Json
+func ToCNCFWorkflow(workflowCR *KogitoServerlessWorkflow) (*cncfmodel.Workflow, error) {
+	if workflowCR == nil {
+		return nil, errors.New("kogitoServerlessWorkflow is nil")
+	}
+	cncfWorkflow := &cncfmodel.Workflow{}
+
+	workflowBytes, err := yaml.Marshal(workflowCR.Spec.Flow)
+	if err != nil {
+		return nil, err
+	}
+	if err = yaml.Unmarshal(workflowBytes, cncfWorkflow); err != nil {
+		return nil, err
+	}
+
+	cncfWorkflow.ID = workflowCR.ObjectMeta.Name
+	cncfWorkflow.Key = workflowCR.ObjectMeta.Annotations[metadata.Key]
+	cncfWorkflow.Name = workflowCR.ObjectMeta.Name
+	cncfWorkflow.Description = workflowCR.ObjectMeta.Annotations[metadata.Description]
+	cncfWorkflow.Version = workflowCR.ObjectMeta.Annotations[metadata.Version]
+	cncfWorkflow.SpecVersion = extractSchemaVersion(workflowCR.APIVersion)
+	cncfWorkflow.ExpressionLang = cncfmodel.ExpressionLangType(extractExpressionLang(workflowCR.ObjectMeta.Annotations))
+
+	return cncfWorkflow, nil
+}
+
+func extractExpressionLang(annotations map[string]string) string {
+	expressionLang := annotations[metadata.ExpressionLang]
+	if expressionLang != "" {
+		return expressionLang
+	}
+	return metadata.DefaultExpressionLang
+}
+
+// Function to extract from the apiVersion the ServerlessWorkflow schema version
+// For example given sw.kogito.kie.org/operatorapi we would like to extract v0.8
+func extractSchemaVersion(version string) string {
+	schemaVersion := path.Base(version)
+	strings.Replace(schemaVersion, "v0", "v0.", 1)
```

Comment:
shouldn't we follow the spect for this field?
use just 0.8, for example instead of v0....